tests/tests_pytorch/strategies/test_ddp_fully_sharded_native.py

```diff
@@ -31,29 +32,27 @@ def test_invalid_on_cpu(tmpdir):
         trainer.strategy.setup_environment()
 
 
-@mock.patch.dict(os.environ, {"CUDA_VISIBLE_DEVICES": "0"})
-@mock.patch("torch.cuda.device_count", return_value=1)
-@mock.patch("torch.cuda.is_available", return_value=True)
 @RunIf(min_torch="1.12dev")
-def test_fsdp_with_sharded_amp(device_count_mock, mock_cuda_available, tmpdir):
-    """Test to ensure that plugin native amp plugin raises Misconfiguration error."""
-    with pytest.raises(
-        MisconfigurationException, match="DDPFullyShardedNativeStrategy currently doesn't support Mixed Precision"
-    ):
-        trainer = Trainer(
-            default_root_dir=tmpdir,
-            fast_dev_run=True,
-            strategy="fsdp_native",
-            accelerator="gpu",
-            devices=1,
-            precision=16,
-        )
-        assert isinstance(trainer.strategy, DDPFullyShardedNativeStrategy)
+@pytest.mark.parametrize("precision, expected", [(16, torch.float16), ("bf16", torch.bfloat16)])
+def test_precision_plugin_config(precision, expected):
+    plugin = FullyShardedNativeMixedPrecisionPlugin(precision=precision, device="cuda")
```

Comment:
Do we have a way to configure only parameters or reduce_dtype? 

i.e., a popular mixed precision approach is to set mixed precision for gradient communication (`reduce_dtype`) only. The config for this would look like:

```
MixedPrecision(reduce_dttype=bfloat16)
```

and param_dtype, config_dtype won't be touched or casted. Could we add support for this as well as this is basically gradient compression?